spanner/client.go

```diff
@@ -557,6 +559,14 @@ func (c *Client) rwTransaction(ctx context.Context, f func(context.Context, *Rea
 		if sh == nil || sh.getID() == "" || sh.getClient() == nil {
 			// Session handle hasn't been allocated or has been destroyed.
 			sh, err = c.idleSessions.take(ctx)
+			if t != nil {
+				// when a batch update operation is called on this transaction, isLongRunningTransaction will be true
```

Comment:
Yes. This doc helps developers understand why this condition was added here.
To give some context, [BatchUpdate](https://github.com/googleapis/google-cloud-go/blob/f609b3cf831fb89c45386f81d0047560120cb3f4/spanner/transaction.go#L1101) method can be called using a ReadWriteTransaction. If there is a retry that happens then a new session will get fetched and if the retry is for a `BatchUpdate` then this new session qualifies to be long running.
If this doc is not added, then there might be confusion that every `ReadWriteTransaction` is not long running (which is not true) and what is the need of this condition here.

Reply:
Yeah, my point is `BatchUpdate` is the only API that uses this currently. But there can be other APIs which will be eligible in future. (For ex - BatchWrite will soon be added in all client libraries, WIP PR in Java - https://github.com/googleapis/java-spanner/pull/2520/). When that's launched in Golang, this piece of documentation will become stale. Because while adding support for new APIs, one would not remember to update the documentation and there's no way to ensure it done. Hence as a general practice, documentation shouldn't include any details that can change in future.

Reply:
Thanks, Arpan. As discussed reworded the comment.